task_geo/data_sources/nasa/nasa.py

```diff
@@ -0,0 +1,214 @@
+
+PARAMETERS = {
+    "temperature": ["T2M", "T2M_MIN", "T2M_MAX"],
+    "humidity": ["RH2M", "QV2M"],
+    "pressure": ["PS"]
+}
+
+COL_NAMES = {
+    "T2M": "avg_temperature",
+    "T2M_MIN": "min_temperature",
+    "T2M_MAX": "max_temperature",
+    "RH2M": "relative_humidity",
+    "QV2M": "specific_humidity",
+    "PS": "pressure"
+}
+
+
+def nasa_data_loc(country, region, sub_region, lat, lon, str_start_date,
+                  str_end_date, parms_str):
+    """
+    Extract data for a single location.
+
+    Parameters
+    ----------
+    country : string
+    region : string
+    sub_region : string
+    lat : string
+    lon : string
+    str_start_date : string
+    str_end_date : string
+    parms_str : string
+
+    Returns
+    -------
+    df : pandas.DataFrame
+
+    """
+    url = f"{base_url}?request=execute&{identifier}&{parms_str}&" \
+          f"startDate={str_start_date}&endDate={str_end_date}&" \
+          f"lat={lat}&lon={lon}&{temporal_average}&{output_format}&" \
+          f"{user_community}&{user}"
+    response = requests.get(url)
+    data_json = response.json()
+    df = pd.DataFrame(data_json['features'][0]['properties']['parameter'])
+    df['country'] = country
+    df['region'] = region
+    df['sub_region'] = sub_region
+    df['lon'] = lon
+    df['lat'] = lat
+    return df
+
+
+def nasa_connector(df_locations, start_date, end_date=None,
+                   parms=["temperature", "humidity", "pressure"],
+                   level="region"):
+    """
+    Retrieve meteorologic data from NASA.
+
+    Given a dataset with columns country, region, sub_region, lon, and lat, for
+    each geographic coordinate (lon, lat) corresponding to a place (specified
+    if country, region, or sub_region) extract the time series of the desired
+    data at the location.
+
+    Arguments:
+    ---------
+        df_locations(pandas.DataFrame): Dataset with columns country, region,
+                                        sub_region, lon, and lat
+        start_date(datetime64): Start date for the time series
+        end_date(pandas.Timestamp): End date fo rthe time series (optional)
+        parms(list of strings): Desired data, accepted are 'temperature',
+                                'humidity', and 'pressure' (optional)
+        level(string):  country, region, or sub_region, tells us at what
+                        granularity we want the data (eg for US sub_region, ie
+                        county, might make more sense than region, ie states,
+                        but in France region could be better, while for
+                        Luxembourg we probably want country) (optional)
+
+    Return:
+    ------
+        pandas.DataFrame:   Columns are country, region, sub_region (non-null),
+                            lon, lat, date, and the desired data.
+    """
+    df_locations = df_locations[~pd.isna(df_locations[level])
+                                & ~pd.isna(df_locations['lon'])
```

Comment:
Ok, couple of formatting issues here:

The first one is the linting issue, which is the [`W503 line break before binary operator`](https://www.flake8rules.com/rules/W503.html), you can solve it by simply put the operators before the line break.

The second one is the indentation issue, you can put your code as:
```python
df_locations =  df_locations[
    ...
]
```
This will give you extra space, so maybe you don't need to break the statement in multiple lines.

The third one is simply about efficient use of pandas, your statement can be changes to:

```python
df_locations = df_locations[~pd.isna(df_locations[[level, 'lon', 'lat']]).all(axis=1)]
```